src/components/widgets/LogsWidget.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect } from 'react';
+import useSWR from 'swr';
+import CodeMirror from '@uiw/react-codemirror';
+import { jupyterTheme } from '@jupyterlab/codemirror';
+import { requestAPI } from '../../handler';
+
+const LogsWidget: React.FC = () => {
+  const { data, mutate } = useSWR('/cluster/logs', { fetcher: (url: string, init: RequestInit) => requestAPI<any>(url, init, false) });
+
+  useEffect(() => {
+    const handle = setInterval(mutate, 700);
```

Comment:
Same as above in similar situation.

Reply:
Same as above; there is no websocket-based comm for server extension.

Reply:
There isn't _yet_. :)

Closing and to be followed in the previous comment.